### PR TITLE
Feat: Make empty gallery customizable through the CMS

### PR DESCRIPTION
**Pull request by AilinMacchia**
## What's the purpose of this pull request?

If you enter a search term that has zero results, the page the user is shown currently links off to categories that are hardcoded in the code.

Empty state page from faststore starter:
image

Hardcoded code with the links, icons and texts:
image

## How it works?

We added Empy Gallery props to Product Gallery section to have dynamic information.

## How to test it?

1. Enter to the store
2. Search a term that won´t match any product available.
3. See the links, texts and icon hardcoded.

### CODE REVIEW

File: packages/core/cms/faststore/sections.json [hunk 1]
```diff
@@ -1607,6 +1607,59 @@
               "default": true
             }
           }
+        },
+        "emptyGallery": {
+          "title": "Empty Gallery",
+          "type": "object",
+          "properties": {
+            "title": {
+              "title": "Icon",
```
Comment (hellofanny) [inline packages/core/cms/faststore/sections.json:1616]:
```suggestion
              "title": "Title",
```

File: packages/core/src/components/ui/ProductGallery/ProductGallery.tsx [hunk 2]
```diff
@@ -67,6 +67,19 @@ export interface ProductGalleryProps {
   }
   sortBySelector?: SortProps
   productCard?: Pick<ProductCardProps, 'showDiscountBadge' | 'bordered'>
+  emptyGallery?: {
+    title?: string
+    topButton?: {
```
Comment (hellofanny) [inline packages/core/src/components/ui/ProductGallery/ProductGallery.tsx:72]: Can we rename it to `firstButton`?

File: packages/core/src/components/ui/ProductGallery/ProductGallery.tsx [hunk 3]
```diff
@@ -67,6 +67,19 @@ export interface ProductGalleryProps {
...
+      text: string
+      href: string
+      iconName: string
+    }
+    lowerButton?: {
```
Comment (hellofanny) [inline packages/core/src/components/ui/ProductGallery/ProductGallery.tsx:77]: Can we rename it to `secondButton`?

File: packages/core/src/components/ui/ProductGallery/ProductGallery.tsx [hunk 4]
```diff
@@ -67,6 +67,19 @@ export interface ProductGalleryProps {
   }
   sortBySelector?: SortProps
   productCard?: Pick<ProductCardProps, 'showDiscountBadge' | 'bordered'>
+  emptyGallery?: {
+    title?: string
+    topButton?: {
+      text: string
+      href: string
+      iconName: string
```
Comment (hellofanny) [inline packages/core/src/components/ui/ProductGallery/ProductGallery.tsx:75]:
```suggestion
      label: string
      url: string
      icon: string
```

File: packages/core/cms/faststore/sections.json [hunk 5]
```diff
@@ -1607,6 +1607,59 @@
...
+                "href": {
+                  "type": "string",
+                  "title": "URL",
+                  "default": "/office"
+                },
+                "iconName": {
+                  "title": "Icon Name",
+                  "type": "string",
+                  "default": "CircleWavyWarning"
+                }
```
Comment (hellofanny) [inline packages/core/cms/faststore/sections.json:1638]:
```suggestion
                "icon": {
                  "title": "Icon",
                  "type": "string",
                  "enumNames": ["Circle Wavy Warning"],
                  "enum": ["CircleWavyWarning"]
                }
```

We just have a set of icons available for now 🙇 So let's keep like that for now, ok?

File: packages/core/src/components/sections/ProductGallery/ProductGallery.tsx [hunk 6]
```diff
@@ -23,6 +23,7 @@ export interface ProductGallerySectionProps {
   loadMorePageButton?: ProductGalleryProps['loadMorePageButton']
   sortBySelector?: ProductGalleryProps['sortBySelector']
   productCard?: ProductGalleryProps['productCard']
+  emptyGalery?: ProductGalleryProps['emptyGallery']
```
Comment (hellofanny) [inline packages/core/src/components/sections/ProductGallery/ProductGallery.tsx:26]:
```suggestion
  emptyGallery?: ProductGalleryProps['emptyGallery']
```

File: packages/core/cms/faststore/sections.json (hunk 1 above)
Comment (AilinMacchia) [reply packages/core/cms/faststore/sections.json]: Ups! my bad

File: packages/core/src/components/ui/ProductGallery/ProductGallery.tsx (hunk 4 above)
Comment (AilinMacchia) [reply packages/core/src/components/ui/ProductGallery/ProductGallery.tsx]: yes sure! Im pretty sure I changed it everywhere

**Comment by hellofanny** [comment]
@AilinMacchia First of all, thanks for contributing! ⭐ I've tested the CMS part, and it's working as expected! Well Done! 🚀 
You started this PR to make the `EmptyGallery` overridable in the `ProductGallery` section, but then you reverted those changes. I was wondering if something happened to change the approach.

**Comment by AilinMacchia** [comment]
hey @hellofanny! Great news everything is working! Yes our first aproach was to have the empty gallery overridable in order to customize it as wanted, but when we made the changes we realized that we were using the product gallery section to add the override but the empty gallery is not called inside that component. So it wasn't working as expected. Maybe we missed something or it is possible making some more changes but ultimately we thought it would be easier to have it be customizable through the CMS for now.

**Comment by hellofanny** [comment]
> Hey @hellofanny!✨ Great news everything is working! Yes our first aproach was to have the empty gallery overridable in order to customize it as wanted, but when we made the changes we realized that we were using the product gallery section to add the override but the empty gallery is not called inside that component. So it wasn't working as expected. Maybe we missed something or it is possible making some more changes but ultimately we thought it would be easier to have it be customizable through the CMS for now.

@AilinMacchia Ohh got it! Does customizing through the CMS fix what you need so far? If so, we can change this PR's title and address the `EmptyGallery` override in another PR. Sounds good?

**Comment by AilinMacchia** [comment]
@hellofanny Yes! the main issue we wanted to achieve here was to be able to change the redirect of the buttons to categories in our tree. I changed the PR name, sorry I didn´t realized I haven´t changed it before 😳. Thank you very much!!

**Comment by hellofanny** [comment]
> @hellofanny Yes! the main issue we wanted to achieve here was to be able to change the redirect of the buttons to categories in our tree. I changed the PR name, sorry I didn´t realized I haven´t changed it before 😳. Thank you very much!!

Niceee! No problem 🥳

**Review by hellofanny — APPROVED** [review]

File: packages/core/cms/faststore/sections.json [hunk 7]
```diff
@@ -1607,6 +1607,61 @@
...
+                "url": {
+                  "type": "string",
+                  "title": "URL",
+                  "default": "/office"
+                },
+                "icon": {
+                  "title": "Icon",
+                  "type": "string",
+                  "enumNames": ["Circle Wavy Warning"],
+                  "enum": ["CircleWavyWarning"]
```
Comment (emersonlaurentino) [inline packages/core/cms/faststore/sections.json:1638]: People may want to use other icons and an enum with just one option is not good practice.

```suggestion
                  "default": "CircleWavyWarning"
```

File: packages/core/src/components/sections/ProductGallery/ProductGallery.tsx [hunk 8]
```diff
@@ -39,7 +40,7 @@ function ProductGallerySection({ ...otherProps }: ProductGallerySectionProps) {
     return (
       <Section className={`${styles.section} section-product-gallery`}>
         <section data-testid="product-gallery" data-fs-product-listing>
-          <EmptyGallery />
+          <EmptyGallery {...otherProps}/>
```
Comment (hellofanny) [inline packages/core/src/components/sections/ProductGallery/ProductGallery.tsx:43]: I missed this part, I think we need to change this part. I'll test it

File: packages/core/src/components/ui/ProductGallery/ProductGallery.tsx [hunk 9]
```diff
@@ -67,6 +67,19 @@ export interface ProductGalleryProps {
   }
   sortBySelector?: SortProps
   productCard?: Pick<ProductCardProps, 'showDiscountBadge' | 'bordered'>
+  emptyGallery?: {
```
Comment (hellofanny) [inline packages/core/src/components/ui/ProductGallery/ProductGallery.tsx:70]: Let's move this to `EmptyGallery` component as `EmptyGalleryProps` interface and import here.

File: packages/core/src/components/sections/ProductGallery/EmptyGallery.tsx [hunk 10]
```diff
@@ -3,40 +3,42 @@ import {
   LinkButton as UILinkButton,
   EmptyState as UIEmptyState,
 } from '@faststore/ui'
+import { ProductGalleryProps } from 'src/components/ui/ProductGallery/ProductGallery'
```
Comment (hellofanny) [inline packages/core/src/components/sections/ProductGallery/EmptyGallery.tsx:7]:
```suggestion
export interface EmptyGalleryProps {
  title?: string
  firstButton?: {
    label: string
    url: string
    icon: string
  }
  secondButton?: {
    label: string
    url: string
    icon: string
  }
}
```

File: packages/core/src/components/sections/ProductGallery/EmptyGallery.tsx [hunk 11]
```diff
@@ -3,40 +3,42 @@ import {
...
 
-function EmptyGallery() {
+function EmptyGallery({ emptyGallery }: ProductGalleryProps) {
+  const { title, firstButton, secondButton } = emptyGallery
```
Comment (hellofanny) [inline packages/core/src/components/sections/ProductGallery/EmptyGallery.tsx:9]:
```suggestion
function EmptyGallery({ title, firstButton, secondButton }: EmptyGalleryProps) {
```

File: packages/core/src/components/sections/ProductGallery/ProductGallery.tsx [hunk 12]
```diff
@@ -23,6 +23,7 @@ export interface ProductGallerySectionProps {
   loadMorePageButton?: ProductGalleryProps['loadMorePageButton']
   sortBySelector?: ProductGalleryProps['sortBySelector']
   productCard?: ProductGalleryProps['productCard']
+  emptyGallery?: ProductGalleryProps['emptyGallery']
 }
 
 function ProductGallerySection({ ...otherProps }: ProductGallerySectionProps) {
```
Comment (hellofanny) [inline packages/core/src/components/sections/ProductGallery/ProductGallery.tsx:31]:
```suggestion
function ProductGallerySection({ emptyGallery, ...otherProps }: ProductGallerySectionProps) {
```

File: packages/core/src/components/sections/ProductGallery/ProductGallery.tsx (hunk 8 above)
Comment (hellofanny) [inline packages/core/src/components/sections/ProductGallery/ProductGallery.tsx]:
```suggestion
          <EmptyGallery {...emptyGallery}/>
```

**Review by hellofanny — CHANGES_REQUESTED** [review]
Hey @AilinMacchia 👋 
I recently re-reviewed your PR :woman-bowing: I think we should make some changes for the `EmptyGalleryProps`. I’m not sure if my suggestions were clear or if they might be a bit confusing.

If you’re okay with it, I can make those changes in your branch, or let me know if you need any help! 

Thanksss 🤩 🤩

**Comment by AilinMacchia** [comment]
> Hey @AilinMacchia 👋 I recently re-reviewed your PR :woman-bowing: I think we should make some changes for the `EmptyGalleryProps`. I’m not sure if my suggestions were clear or if they might be a bit confusing.
> 
> If you’re okay with it, I can make those changes in your branch, or let me know if you need any help!
> 
> Thanksss 🤩 🤩

Yes sure!🤗 Feel free to make the changes you want to this branch! If you need me to do do something please let me know 🤩

**Review by hellofanny — APPROVED** [review]

File: packages/core/src/components/ui/ProductGallery/ProductGallery.tsx [hunk 13]
```diff
@@ -67,6 +68,7 @@ export interface ProductGalleryProps {
   }
   sortBySelector?: SortProps
   productCard?: Pick<ProductCardProps, 'showDiscountBadge' | 'bordered'>
+  emptyGallery?: EmptyGalleryProps
```
Comment (eduardoformiga) [inline packages/core/src/components/ui/ProductGallery/ProductGallery.tsx:71]: hmm, we don't use this prop here in the `ui/ProductGallery`. How about remove this line and its import and use it directly in https://github.com/AilinMacchia/faststore/blob/feature/empty-gallery-override/packages/core/src/components/sections/ProductGallery/ProductGallery.tsx#L28C27-L28C27 ?

**Review by eduardoformiga — APPROVED** [review]
Thank you for contributing 😄 

LGTM! I just left a comment but it's not blocking. I'll let it approved.

**Pull request merged**